core/src/contexts.ts

```diff
@@ -101,91 +112,91 @@ export function validateVendorParts(parts: Array<string>): boolean {
 }
 
 export function validateVendor(input: string): boolean {
-    let parts = input.split('.');
+    const parts = input.split('.');
     if (parts && parts.length > 1)
         return validateVendorParts(parts);
     return false;
 }
 
 export function getRuleParts(input: string): Array<string> | undefined {
-    const re = new RegExp('^iglu:((?:(?:[a-zA-Z0-9-_]+|\\*)\.)+(?:[a-zA-Z0-9-_]+|\\*))\/([a-zA-Z0-9-_.]+|\\*)\/jsonschema\/([1-9][0-9]*|\\*)-(0|[1-9][0-9]*|\\*)-(0|[1-9][0-9]*|\\*)$');
-    let matches = re.exec(input);
+    const re = new RegExp('^iglu:((?:(?:[a-zA-Z0-9-_]+|\\*).)+(?:[a-zA-Z0-9-_]+|\\*))/([a-zA-Z0-9-_.]+|\\*)/jsonschema/([1-9][0-9]*|\\*)-(0|[1-9][0-9]*|\\*)-(0|[1-9][0-9]*|\\*)$');
+    const matches = re.exec(input);
     if (matches !== null && validateVendor(matches[1]))
         return matches.slice(1,6);
     return undefined;
 }
 
 export function isValidRule(input: string): boolean {
-    let ruleParts = getRuleParts(input);
+    const ruleParts = getRuleParts(input);
     if (ruleParts) {
-        let vendor = ruleParts[0];
+        const vendor = ruleParts[0];
         return ruleParts.length === 5 && validateVendor(vendor);
     }
     return false;
 }
 
-export function isStringArray(input: any): boolean {
+export function isStringArray(input: unknown): input is Array<string> {
     return Array.isArray(input) && input.every((x) => { return typeof x === 'string' });
 }
 
-export function isValidRuleSetArg(input: any): boolean {
+export function isValidRuleSetArg(input: unknown): boolean {
     if (isStringArray(input))
-        return input.every((x) => { return isValidRule(x) });
+        return input.every((x: string) => { return isValidRule(x) });
     else if (typeof input === 'string')
         return isValidRule(input);
     return false;
 }
 
-export function isSelfDescribingJson(input: any) : boolean {
-    if (isNonEmptyJson(input))
-        if ('schema' in input && 'data' in input)
-            return (typeof(input.schema) === 'string' && typeof(input.data) === 'object');
+export function isSelfDescribingJson(input: unknown): input is SelfDescribingJson {
+    const sdj = input as SelfDescribingJson;
+    if (isNonEmptyJson(sdj))
+        if ('schema' in sdj && 'data' in sdj)
+            return (typeof(sdj.schema) === 'string' && typeof(sdj.data) === 'object');
     return false;
 }
 
-export function isEventJson(input: any) : boolean {
-    if (isNonEmptyJson(input) && ('e' in input))
-        return (typeof(input.e) === 'string');
+export function isEventJson(input: unknown): input is PayloadDictionary {
+    const payload = input as PayloadDictionary;
+    if (isNonEmptyJson(payload) && ('e' in payload))
+        return (typeof(payload.e) === 'string');
     return false;
 }
 
-export function isRuleSet(input: any) : boolean {
+export function isRuleSet(input: unknown): input is Record<string, unknown> {
+    const ruleSet = input as Record<string, unknown>;
```

Comment:
This looks like one of the few places where the code is changed, rather than just superficial changes.

Previously there was a `if (isPlainObject(input))` whereas here you cast it to a Record.  Please just confirm you are satisfied this is always a plain object?

Reply:
Ah yeah, I think I removed this in the reshuffle. I don't think we can be sure this is always a plan object as the input comes from the user of the tracker (typically). I'll add this back in.